### PR TITLE
Fix - move device_id from run_device to run_metrics

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
Closes https://github.com/Clinical-Genomics/add-new-tech/issues/20. Adds an alembic revision removing column `device_id` from table`run_device` and adding it to `run_metrics` table as a foreign key.

### Added
- Foreign key to `run_metrics`

### Changed

- Removed the column `device_id` from table`run_device` and adding it to `run_metrics` table.

### How to prepare for test

- [ ] Stablish an SSH connection to cg stage

### How to test

- [ ] Verify that the current alembic version is `4766684153e3`
- [ ] Do `alembic -c $ALEMBIC_CG_STAGE upgrade e5c6bf847e25` and verify that the tables are changed
- [ ] Do `alembic -c $ALEMBIC_CG_STAGE downgrade 4766684153e3` and verify that the tables return to their original state.

### Expected test outcome

- [x] Check that after the upgrade the `run_metrics` table has a column named `device_id` and that it is a foreign key
- [x] Check that after the downgrade that the changes are reverted

## Review

- [x] Tests executed by SD
- [x] "Merge and deploy" approved by CO, SA, VJ
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+"""
+
+import sqlalchemy as sa
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "078884a6b2d2"
+down_revision = "9b188aee9577"
```
Comment (seallard) [inline alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py:15]: Make sure this one is correct before merging!

File: alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.drop_column("run_device", "device_id")
+    op.add_column("run_metrics", sa.Column("device_id", sa.Integer(), nullable=False))
+    op.create_foreign_key("fk_device_id", "run_metrics", "run_device", ["device_id"], ["id"])
```
Comment (ChrOertlin) [inline alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py:23]: I think I didnt create foreign keys ont he other tables either, should we add the samplerun etrics keys as well?=

**Review by ChrOertlin — APPROVED** [review]
Looks good

File: alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py (hunk 2 above)
Comment (seallard) [inline alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py:23]: I think this one is correct, might be easier if named arguments are used.

You are adding a foreign key on the run metrics table referring to the run device table.

**Review by seallard — APPROVED** [review]
👍

**Comment by diitaz93** [comment]
> I think this will require a change in the python models as well! But looks promising!

They python models are currently showing these columns, so this PR fixes the db to match the python models 🙂

File: alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py (hunk 2 above)
Comment (diitaz93) [reply alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py]: yes, I changed it to @islean suggestion to add everything in one line

[the remark this answers is not in the document]
Comment (diitaz93) [reply alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py]: done

[the remark this answers is not in the document]
Comment (diitaz93) [reply alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py]: It actually looks better in two commands, see now @islean

Comment (diitaz93) [reply alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py]: reverted

File: alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py (hunk 1 above)
Comment (diitaz93) [reply alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py:15]: last revision in #3202 had `9b188aee9577`

**Review by Vince-janv — APPROVED** [review]
Looks good. And you have tested the two functions?

File: alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py (hunk 2 above)
Comment (Vince-janv) [reply alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py]: Feel free to add it in a separate PR

File: alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py (hunk 1 above)
Comment (diitaz93) [reply alembic/versions/2024_05_13_078884a6b2d2_fix_new_tech_tables.py]: correction, last revision is `4766684153e3`

**Comment by diitaz93** [comment]
> Looks good. And you have tested the two functions?

Upgrade and downgrade are passing tests now

**Pull request merged**